tasks/auditd.yml

```diff
@@ -18,41 +18,14 @@
 
 - name: set RedHat audit grub cmdline
   become: 'yes'
-  lineinfile:
-    line: 'GRUB_CMDLINE_LINUX="$GRUB_CMDLINE_LINUX {{ grub_audit_cmdline }} {{ grub_audit_backlog_cmdline }}"'
-    dest: /etc/default/grub
-    insertafter: '^GRUB_CMDLINE_LINUX="[a-z]+'
-    state: present
-    create: 'no'
-    mode: 0644
-    owner: root
-    group: root
+  command: grubby --update-kernel=ALL --args="{{ grub_audit_cmdline }} {{ grub_audit_backlog_cmdline }}"
   notify:
     - update grub2
   when: ansible_os_family == "RedHat"
   tags:
     - auditd
     - grub
 
-- name: check RedHat auditd ExecStartPost
-  environment:
-    PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
-  command: grep -qo "^ExecStartPost=-/sbin/auditctl -R" /usr/lib/systemd/system/auditd.service
-  register: execstartpost_auditd_set
-  when: ansible_os_family == "RedHat"
-  changed_when: execstartpost_auditd_set.rc != 0
-  failed_when: execstartpost_auditd_set.rc > 1
-
-- name: set audit service execstartpost
```

Comment:
why is this removed?

Reply:
Used only in next removed task:  name: set audit service execstartpost

I have provided explanation in PR first comment,...

Reply:
why not? https://github.com/konstruktoid/ansible-role-hardening/blob/master/tasks/auditd.yml#L37-L55 ensures that the current rules are loaded when `auditd` is (re)started.

Reply:
1. The rule modifies the **/usr/lib/systemd/system/auditd.service** Then any updat will kill the change 
2. in the rules is final rule **-e 2**  forces audit to be put in immutable mode. Audit changes can only be made on system reboot.
3. Seems that staert the audit service again forces rules re-generation... 

By my opinion, it's obsolete, and if not, then it's implemented on the wrong place

Reply:
Yeah, `/usr/lib/systemd/system/auditd.service` is the incorrect file. It should be placed in `/etc`.

Update (the wrong) .service file at https://github.com/konstruktoid/ansible-role-hardening/blob/master/tasks/auditd.yml#L50, add our rules at https://github.com/konstruktoid/ansible-role-hardening/blob/master/tasks/auditd.yml#L165 and then generate and restart `auditd` if file has been changed at https://github.com/konstruktoid/ansible-role-hardening/blob/master/handlers/main.yml#L2-L12.  
This is to ensure our rules are in place and used.

Reply:
>     3. Seems that staert the audit service again forces rules re-generation...
where?

Reply:
Sorry. Point3 is wrong. but with -e 2 as last rule, you can reload, restart, re-generat rules, but without any efect. You have to reboot the machine to modify the rule

Reply:
yeah, should probably use a `when:` if `-e 2` is set

Reply:
will fix those tasks after merge so you dont have to mess around with rebasing